python/lsst/dax/data_generator/columns.py

```diff
@@ -28,179 +203,308 @@ def __call__(self, cell_id, length):
+            blocks["entire"] = mergeBlocks(blockNS, blockEW)
+            if not edge_only:
+                box_middle = SimpleBox(boxes["east"].raB, boxes["west"].raA,
+                                       boxes["north"].decB, boxes["south"].decA)
+                block_middle = self._generateBlock(chunk_id, box_middle, sub_length)
+                blocks["entire"] = mergeBlocks(blocks["entire"], block_middle)
+        return blocks["entire"]
 
 
 class ObjIdGenerator(ColumnGenerator):
 
-    def __call__(self, cell_id, length, **kwargs):
+    def __call__(self, chunk_id, length, seed, **kwargs):
         """
         Returns
         -------
         object_id : array
             Array containing unique IDs for each object
         """
 
-        return (cell_id * 100000) + np.arange(length)
+        # TODO: more than 100k objects in a chunk will cause issues
+        #       Replace 100000 with max_objects_per_chunk?
+        return (chunk_id * 100000) + np.arange(length)
 
 
 class VisitIdGenerator(ColumnGenerator):
 
-    def __call__(self, cell_id, length, **kwargs):
+    def __call__(self, chunk_id, length, seed, **kwargs):
         """
         Returns
         -------
         visit_id : array
             Array containing unique IDs for each visit
         """
 
-        return 10000000000 + (cell_id * 100000) + np.arange(length)
+        # TODO: This shouldn't have the same issue as objects/chunk
+        #       but maybe replace 100000 with max_objects_per_chunk?
+        return 10000000000 + (chunk_id * 100000) + np.arange(length)
 
 
 class MagnitudeGenerator(ColumnGenerator):
     """
+    Parameters
+    ----------
+    n_mags : int
+        Number of magnitude columns to make.
+    min_mag : float
+        Minimum value for generated magnitudes.
+    max_mag : float
+        Maximum value for generated magnitudes.
+    column_val : int
+        Arbitrary integer that should be different from other
+        column values. Used in random number generation.
+
+    Note
+    ----
     Currently generates a flat magnitude distribution. Should properly
     be some power law.
+    If there is more than one call to this in a single table, there will
+    be correlation between rows as the same random numbers will be used.
     """
 
-    def __init__(self, n_mags=1, min_mag=0, max_mag=27.5):
+    def __init__(self, n_mags=1, min_mag=0, max_mag=27.5, column_val=7):
         self.n_mags = n_mags
         self.min_mag = min_mag
         self.max_mag = max_mag
+        self.column_val = column_val  # arbitrary, but different from other columns
 
-    def __call__(self, cell_id, length, **kwargs):
-        mags = []
+    def __call__(self, chunk_id, length, seed, **kwargs):
+
+        np.random.seed(calcSeedFrom(chunk_id, seed, self.column_val))
+
+        # This needs to be made row by row not column by column, as
+        # row by row results in repeatable values when doing edges first.
```

Comment:
Generating it by column requires knowing how many rows are in the edge only version, and there's no mechanism for that at this point. With luck this will be fast enough.